File: program/find-sum-of-numbers-by-recursion/FindSumOfNumbersByRecursion.cs
using System;

public class FindSumOfNumbersByRecursion {

    public int sumOfNumbersByRecursion(int n) {
        int sum;
        if (n == 1){
            return 1;
        }
        else{
            sum = sumOfNumbersByRecursion(n - 1) + n;
            return sum;
        }
    }
    public static void Main(String[] args){
        FindSumOfNumbersByRecursion n = new FindSumOfNumbersByRecursion();
        Console.Write("Input  : ");
        for(int i = 1; i <= 5; i++){
            Console.Write(i + " ");
        }
        Console.WriteLine(" ");
        Console.WriteLine("Output : {0}", n.sumOfNumbersByRecursion(5));

    }
}
        

File: program/program/find-sum-of-numbers-by-recursion/find_sum_of_numbers_by_recursion.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func sumRecursive(num int) int {
	if num == 0 {
		return 0
	}
	return num%10 + sumRecursive(num/10)
}

func main() {
	var input string

	fmt.Println("Input :")
	fmt.Scanln(&input)

	numberStrings := strings.Fields(input)
	numbers := make([]int, len(numberStrings))

	for i, numStr := range numberStrings {
		num, _ := strconv.Atoi(numStr)
		numbers[i] = num
	}

	sum := 0
	for _, num := range numbers {
		sum += sumRecursive(num)
	}
	fmt.Println("Output :", sum)
}